File: Grocery-Backend/frontend/src/admin/services/userService.js
import axiosInstance from './axiosInstance';

export const userService = {
  getAllUsers: async () => {
    const res = await axiosInstance.get('/users');
    return res.data;
  },

  deleteUser: async (userId) => {
    const res = await axiosInstance.delete(`/users/${userId}`);
    return res.data;
  }
};

File: Grocery-Backend/src/main/java/com/grocery/service/UserService.java
package com.grocery.service;

import java.util.List;

import com.grocery.dto.UserRegistrationRequest;
import com.grocery.dto.UserResponse;
import com.grocery.dto.UserUpdateRequest;

public interface UserService {
	UserResponse registerUser(UserRegistrationRequest request);

	UserResponse getUserById(Long userId);

	List<UserResponse> getAllUsers();

	void deleteUser(Long userId);
	
	UserResponse updateUser(Long userId, UserUpdateRequest request);

	UserResponse getMyProfile();
}
